src/webui/components/Search/index.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @prettier
+ * @flow
+ */
+
+import React, { Component } from 'react';
+
+import { default as IconSearch } from '@material-ui/icons/Search';
+import InputAdornment from '@material-ui/core/InputAdornment';
+
+import API from '../../utils/api';
+import AutoComplete from '../AutoComplete';
+import colors from '../../utils/styles/colors';
+import { getDetailPageURL } from '../../utils/url';
+
+import { AutoCompleteWrapper } from './styles';
+
+import { IProps, IState } from './types';
+class Search extends Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+
+    this.state = {
+      search: '',
+      suggestions: [],
+      loading: false,
+      loaded: false,
+      error: false,
+    };
+  }
+
+  handlePackagesClearRequested = () => {
+    this.setState({
+      suggestions: [],
+    });
+  };
+
+  // eslint-disable-next-line no-unused-vars
+  handleSearch = (event: SyntheticKeyboardEvent<HTMLInputElement>, { newValue, method }: { newValue: string, method: string }) => {
+    const value = newValue.trim();
+    this.setState({ search: value });
+  };
+
+  // eslint-disable-next-line no-unused-vars
+  handleClickSearch = (event: SyntheticKeyboardEvent<HTMLInputElement>, { suggestionValue, method }: { suggestionValue: any[], method: string }) => {
```

Comment:
This line seems to long

Reply:
yes!...and we should define an interface for suggestionValue...for now it's ok the any type, but we should improve it..